pkg/registry/apis/provisioning/list.go

```diff
@@ -0,0 +1,99 @@
+package provisioning
+
+import (
+	"context"
+	"log/slog"
+	"net/http"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/registry/rest"
+
+	provisioning "github.com/grafana/grafana/pkg/apis/provisioning/v0alpha1"
+)
+
+type listConnector struct {
+	repoGetter RepoGetter
+	logger     *slog.Logger
+}
+
+func (*listConnector) New() runtime.Object {
+	// This is added as the "ResponseType" regardless what ProducesObject() returns
+	return &provisioning.ResourceWrapper{}
+}
+
+func (*listConnector) Destroy() {}
+
+func (*listConnector) NamespaceScoped() bool {
+	return true
+}
+
+func (*listConnector) GetSingularName() string {
+	return "List"
+}
+
+func (*listConnector) ProducesMIMETypes(verb string) []string {
+	return []string{"application/json"}
+}
+
+func (c *listConnector) ProducesObject(verb string) any {
+	return c.New()
+}
+
+func (*listConnector) ConnectMethods() []string {
+	return []string{http.MethodGet, http.MethodPost}
+}
+
+func (*listConnector) NewConnectOptions() (runtime.Object, bool, string) {
+	return nil, false, ""
+}
+
+func (c *listConnector) Connect(
+	ctx context.Context,
+	name string,
+	opts runtime.Object,
+	responder rest.Responder,
+) (http.Handler, error) {
+	logger := c.logger.With("repository_name", name)
+	repo, err := c.repoGetter.GetRepository(ctx, name)
+	if err != nil {
+		return nil, err
+	}
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		query := r.URL.Query()
+		ref := query.Get("ref")
+		if ref != "" {
+			logger = logger.With("ref", ref)
+		}
+		continueToken := query.Get("continueToken")
```

Comment:
what's this for? for recursion?

Reply:
it is the k8s standard approach for list pagination.  The response meta has a continueToken that you send in the next request.

BUT we are currently always returning eveyrthing so I can skip it